### ISSUE TITLE
Traces DS 3.2

### ISSUE THREAD

**Issue by mbezuljTT** [tenstorrent/tt-metal#46451]

**Comment by nmilicevicTT** [comment]
DeepSeek-V3.2 debug trace — results/deepseek-v32/

What it is: a per-token GPU golden trace of one 5120-token prefill (single forward) of deepseek-ai/DeepSeek-V3.2, captured from vLLM (0.22.1rc1.dev346+gfe1d923af) at the module boundaries of the DSA (DeepSeek Sparse Attention) block. Captured rank-0 of an 8-way tensor-parallel run, enforce_eager=True. All streams are TP-replicated → full tensors, not shards. Total size: 30 GB.

Config (from metadata.json): 61 layers (0–2 dense, 3–60 MoE) · hidden 7168 · 256 experts / top-8 (grouped: n_group=8, topk_group=4) · MLA kv_lora_rank=512, qk_nope=128, qk_rope=64, v_head=128 · DSA index_topk=2048, index_n_heads=64.

Row axis of every per-token stream = seq_len = 5120. Manifests index.json + metadata.json are plain files (not LFS).

<img width="1271" height="701" alt="Image" src="https://github.com/user-attachments/assets/3628467a-1306-44da-ae3e-24a12648e730" />


trace repo:

```
GIT_LFS_SKIP_SMUDGE=1 git clone https://github.com/tenstorrent/bit_sculpt.git
cd bit_sculpt
GIT_LFS_SKIP_SMUDGE=1 git switch --track origin/nmilicevic/v3_2-trace
```

this fetches references to the files.

to download trace for layers 0, 30, 60 (or specify some other) run:

```
RUN=results/deepseek-v32
INC=""
for L in 0 30 60; do
  for s in module_io/mla_input_layer module_io/mla_output_layer module_io/indexer_input_layer \
           dsa/indexer_logits_layer dsa/dsa_topk_indices_layer; do
    INC="$INC,$RUN/${s}_${L}/*"
  done
  INC="$INC,$RUN/kv_cache/layer_${L}/*"
done
git lfs pull --include="${INC#,}"
```

then you can load and use those files:

```
from safetensors.torch import load_file
import glob, torch

def load_stream(stream_dir):
    """Load one (stream, layer) → full tensor. Single-file or chunked, both work."""
    files = sorted(glob.glob(f"{stream_dir}/rows_*.safetensors"))
    parts = []
    for f in files:
        d = load_file(f)
        (key,) = d.keys()          # exactly one tensor per file
        parts.append(d[key])
    return torch.cat(parts, dim=0)

run = "results/deepseek-v32"
L = 0

# e.g. these, make sure you git lfs pulled those layers
mla_in  = load_stream(f"{run}/module_io/mla_input_layer_{L}")     # (5120, 7168) bf16
mla_out = load_stream(f"{run}/module_io/mla_output_layer_{L}")    # (5120, 7168) bf16
idx_in  = load_stream(f"{run}/module_io/indexer_input_layer_{L}") # (5120, 7168) bf16  (== mla_in)
logits  = load_stream(f"{run}/dsa/indexer_logits_layer_{L}")      # (5120, 5120) fp32; logits.to(torch.bfloat16) 
topk    = load_stream(f"{run}/dsa/dsa_topk_indices_layer_{L}")    # (5120, 2048) int32
kv      = load_stream(f"{run}/kv_cache/layer_{L}")                # (5120, 576)  bf16
```
